catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/BrickAdapter.kt

```diff
@@ -148,9 +148,18 @@ class BrickAdapter(private val sprite: Sprite) :
         item.checkBox.setOnClickListener { onCheckBoxClick(position) }
         when (checkBoxMode) {
             NONE -> handleCheckBoxModeNone(item)
-            CONNECTED_ONLY -> handleCheckBoxModeConnectedOnly(item, itemView, position)
-            ALL -> handleCheckBoxModeAll(item)
-            SCRIPTS_ONLY -> handleCheckBoxModeScriptsOnly(item)
+            CONNECTED_ONLY -> {
+                itemView.setOnClickListener { onCheckBoxClick(position) }
+                handleCheckBoxModeConnectedOnly(item, itemView, position)
+            }
+            ALL -> {
+                itemView.setOnClickListener { onCheckBoxClick(position) }
+                handleCheckBoxModeAll(item)
+            }
+            SCRIPTS_ONLY -> {
+                itemView.setOnClickListener { onCheckBoxClick(position) }
+                handleCheckBoxModeScriptsOnly(item)
+            }
```

Comment:
- set item.onClickListener before when statement to reduce code
- reset onClickListener when checkBoxMode is NONE

```suggestion
            NONE -> {
                itemView.setOnClickListener { onItemClickListener?.onBrickClick(item, position) }
                handleCheckBoxModeNone(item)
            }

            CONNECTED_ONLY -> handleCheckBoxModeConnectedOnly(item, itemView, position)
            ALL -> handleCheckBoxModeAll(item)
            SCRIPTS_ONLY -> handleCheckBoxModeScriptsOnly(item)
```